.github/actions/setup_dfx/action.yml

```diff
@@ -13,9 +13,11 @@ runs:
               echo "dfx-version=${DFX_VERSION}" >> "$GITHUB_OUTPUT"
           shell: bash
 
+        # Install dfx (Note: dfx must be installed before `npx azle` because the azle installation process requires dfx)
         - name: Install dfx
-          run: |
-              # Install dfx (Note: dfx must be installed before `npx azle` because the azle installation process requires dfx)
-              src/build/stable/commands/install_global_dependencies/install_dfx.sh "${{ steps.get-dfx-version.outputs.dfx-version }}"
-              echo "$HOME/.local/share/dfx/bin" >> $GITHUB_PATH
+          uses: ./.github/actions/retry_command
+          with:
+              command: src/build/stable/commands/install_global_dependencies/install_dfx.sh "${{ steps.get-dfx-version.outputs.dfx-version }}"
+
+        - run: echo "$HOME/.local/share/dfx/bin" >> $GITHUB_PATH
```

Comment:
Isn't this already taken care of in the install_dfx.sh script? And won't it explicitly fail if this doesn't work?

Reply:
My memory is that the context doesn't last outside of the script. That's the reason we had this outside the script in the first place